src/main/java/org/verdictdb/sqlsyntax/ImpalaSyntax.java

```diff
@@ -114,6 +114,15 @@ public String randFunction() {
     return String.format("rand(unix_timestamp()+%d)", randomNum);
   }
 
+  @Override
+  public String rowNumberFunction(String partition) {
+    if (partition.equals("")) {
+      return "ROW_NUMBER() OVER ()";
+    } else {
+      return String.format("ROW_NUMBER() OVER (PARTITION BY %s ORDER BY %s)", partition, randFunction());
```

Comment:
Why `ORDER BY rand()` only in Impala? Is it randomly ordered for other DBMS by default?

Reply:
Yes. At least in the current version, Impala require order by clause.

Reply:
So all the other DBMS do not require `ORDER BY` and they randomly order within the specified partitions by default? I just want to make sure.

Reply:
Yes, I think Sqlite need ORDER BY as well. But I just find Sqlite syntax hasn't been implemented yet.